tests/testthat/test-shinytest2-tm_g_scatterplot.R

```diff
@@ -0,0 +1,194 @@
+  testthat::expect_null(app$get_active_module_input("col_facet-dataset_ADSL_singleextract-select"))
+  testthat::expect_equal(app$get_active_module_input("alpha"), 1)
+  testthat::expect_equal(app$get_active_module_input("shape"), "circle")
+  testthat::expect_equal(app$get_active_module_input("color"), "#000000")
+  testthat::expect_equal(app$get_active_module_input("size"), 5)
+  testthat::expect_false(app$get_active_module_input("rotate_xaxis_labels"))
+  testthat::expect_false(app$get_active_module_input("add_density"))
+  testthat::expect_false(app$get_active_module_input("rug_plot"))
+  testthat::expect_false(app$get_active_module_input("show_count"))
+  testthat::expect_null(app$get_active_module_input("smoothing_degree"))
+  testthat::expect_equal(app$get_active_module_input("ggtheme"), "gray")
+
+  app$stop()
+})
+
+testthat::test_that("e2e - tm_g_scatterplot: base for the log transformation can be applied", {
+  skip_if_too_deep(5)
+
+  app <- app_driver_tm_g_scatterplot()
+
+  app$set_active_module_input("log_x", TRUE)
+  app$set_active_module_input("log_x_base", "log2")
+  app$set_active_module_input("log_y", TRUE)
+  app$set_active_module_input("log_y_base", "log10")
+  app$expect_no_validation_error()
+
+  app$stop()
+})
+
+testthat::test_that("e2e - tm_g_scatterplot: base for the log transformation can be applied", {
+  skip_if_too_deep(5)
+
+  app <- app_driver_tm_g_scatterplot()
+
+  app$set_active_module_input("log_x", TRUE)
+  app$set_active_module_input("log_x_base", "log2")
+  app$set_active_module_input("log_y", TRUE)
+  app$set_active_module_input("log_y_base", "log10")
+  app$expect_no_validation_error()
+
+  app$stop()
+})
+
+testthat::test_that("e2e - tm_g_scatterplot: log transform is only possible for positive numeric vars", {
+  skip_if_too_deep(5)
+
+  app <- app_driver_tm_g_scatterplot()
+
+  app$set_active_module_input("x-dataset_ADSL_singleextract-select", "BMRKR2")
+  app$set_active_module_input("log_x", TRUE)
+  app$expect_validation_error()
+
+  app$set_active_module_input("x-dataset_ADSL_singleextract-select", "BMRKR1")
+  app$expect_no_validation_error()
+
+  app$set_active_module_input("y-dataset_ADSL_singleextract-select", "BMRKR2")
+  app$set_active_module_input("log_y", TRUE)
+  app$expect_validation_error()
+
+  app$stop()
+})
+
+testthat::test_that("e2e - tm_g_scatterplot: get validation error when facetting with the same row & col variable", {
+  skip_if_too_deep(5)
+
+  app <- app_driver_tm_g_scatterplot()
+
+  app$set_active_module_input("row_facet-dataset_ADSL_singleextract-select", "RACE")
+  app$set_active_module_input("col_facet-dataset_ADSL_singleextract-select", "RACE")
+  app$expect_validation_error()
+
+  app$stop()
+})
+
+testthat::test_that("e2e - tm_g_scatterplot: encoding inputs produce output without validation errors", {
+  skip_if_too_deep(5)
+
+  app <- app_driver_tm_g_scatterplot()
+
+  app$set_active_module_input("color_by-dataset_ADSL_singleextract-select", "REGION1")
```

Comment:
Consider adding more `app$expect_no_validation_error()` to make sure that all states are error-free

Same here, follow the same strategy as the other comments :-)